MdePkg/Test/Library/MockPeiServicesLib/MockPeiServicesLib.c

```diff
@@ -0,0 +1,42 @@
+/** @file
+  Mock implementation of the Pei Services Library.
+  Copyright (C) Microsoft Corporation.
+  SPDX-License-Identifier: BSD-2-Clause-Patent
+**/
+
+#include <stdio.h>
+#include <string.h>
+#include <stdarg.h>
+#include <stddef.h>
+#include <setjmp.h>
+#include <cmocka.h>
+
+#include <Uefi.h>
+#include <PiPei.h>
+#include <Library/UnitTestLib.h>
+#include <MockPeiServicesLib.h>
+
+/**
+  This service enables PEIMs to discover a given instance of an interface.
+  @param  Guid                  A pointer to the GUID whose corresponding interface needs to be
+                                found.
+  @param  Instance              The N-th instance of the interface that is required.
+  @param  PpiDescriptor         A pointer to instance of the EFI_PEI_PPI_DESCRIPTOR.
+  @param  Ppi                   A pointer to the instance of the interface.
+  @retval EFI_SUCCESS           The interface was successfully returned.
+  @retval EFI_NOT_FOUND         The PPI descriptor is not found in the database.
+**/
+EFI_STATUS
+EFIAPI
+PeiServicesLocatePpi (
+  IN CONST EFI_GUID              *Guid,
+  IN UINTN                       Instance,
+  IN OUT EFI_PEI_PPI_DESCRIPTOR  **PpiDescriptor  OPTIONAL,
+  IN OUT VOID                    **Ppi
+  )
+{
+  PPI_STATUS  *PpiStatus = (PPI_STATUS *)mock ();
+
+  *Ppi = PpiStatus->Ppi;
+  return PpiStatus->Status;

```

Comment:
This is one I wrote for a specific case in SetupDataPkg, this can either be propagated to the common code as a pattern for other unit tests to use or we can get rid of the `PPI_STATUS` structure and have two separate mocks here. No preference, just something to keep in mind if other unit tests mock this function.